base/errorshow.jl

```diff
@@ -1086,6 +1092,28 @@ end
 
 Experimental.register_error_hint(methods_on_iterable, MethodError)
 
+# Display a hint in case the user tries to access non-member fields of container type datastructures
+function fielderror_hint_handler(io, exc)
+    @nospecialize
+    field = exc.field
+    type = exc.type
+    if type <: Dict
+        println(io,
+            """
+            \nDid you mean to access dict values using key: `$field` ?
+            Consider using `indexing` operation.
+            Example:
+                ```julia
+                dict = Dict($(field)=>someValue)
```

Comment:
```suggestion
                dict = Dict(:$(field)=>someValue)
```
`Dict(a=>b)` refers to the value of the variable `a`, not the symbol `:a`.